packages/contracts-bedrock/src/L2/CrossL2Inbox.sol

```diff
@@ -159,18 +159,20 @@ contract CrossL2Inbox is ISemver, TransientReentrancyAware {
         }
     }
 
+    bytes32 constant MSB_MASK = 0x00ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff;
+
     bytes32 constant TYPE_3_MASK = 0x03ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff;
 
-    function calculateChecksum(Identifier memory _id, bytes32 _msgHash) public pure returns (bytes32) {
+    function calculateChecksum(Identifier memory _id, bytes32 _msgHash) public pure returns (bytes32 checksum_) {
         bytes32 logHash = keccak256(abi.encodePacked(_id.origin, _msgHash));
 
-        bytes memory idPacked = abi.encodePacked(_id.logIndex, _id.timestamp, _id.blockNumber);
+        bytes memory idPacked = abi.encode(_id.logIndex, _id.timestamp, _id.blockNumber);
 
         bytes32 idLogHash = keccak256(abi.encodePacked(logHash, idPacked));
 
         bytes32 bareChecksum = keccak256(abi.encodePacked(idLogHash, _id.chainId));
 
-        return bareChecksum & TYPE_3_MASK;
+        checksum_ = (bareChecksum & MSB_MASK) | TYPE_3_MASK;
```

Comment:
The `|` here sets all bits of the checksum to 1, making the checksum always the same value.
This is difficult low-level code, not for regular solidity. Should we maybe just use the yul code from the spec?
Doing a `mstore8` to overwrite a single byte to me is a lot more readable than manual bit-masking and or-ing.